packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -74,19 +74,42 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
         this.root.resolve(root);
     }
 
-    getRoot(): Promise<string | undefined> {
+    getWorkspace(): Promise<string | undefined> {
         return this.root.promise;
     }
 
-    async setRoot(uri: string): Promise<void> {
+    async setWorkspace(uri: string): Promise<void> {
         this.root = new Deferred();
+        const listUri: string[] = [];
+        const oldListUri = await this.getRecentWorkspaces();
+        listUri.push(uri);
+        if (oldListUri) {
+            oldListUri.forEach(element => {
+                if (element !== uri && element.length > 0) {
```

Comment:
A low-effort way of pruning the list would be to discard any workspace directory that does not exist on disk anymore. It could easily be checked here.

Reply:
Done, filter to return only the existing workspaces. On the next saving, will only save valid workspaces